src/stretch/mapping/voxel/voxel_map.py

```diff
@@ -272,7 +273,18 @@ def is_valid(
         y1 = y0 + dim
 
         if obstacles is None:
-            obstacles, explored = self.voxel_map.get_2d_map()
+            attempt = 0
+            max_attempts = 10
+            while True:
+                try:
+                    obstacles, explored = self.voxel_map.get_2d_map()
+                    break
+                except Exception as e:
```

Comment:
Try-catch tends to be slow; what exceptions happen here?

Reply:
Sometimes voxel_map is empty and the method call fails

Reply:
Better to change the function to handle the error properly. try-except is very slow in python